### PR TITLE
Fixes #30577 - use host importer

### PR DESCRIPTION
**Pull request by lzap**
Fixing tests after https://github.com/theforeman/foreman/pull/7749

### CODE REVIEW

File: app/services/foreman_discovery/host_fact_importer.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class ForemanDiscovery::HostFactImporter < ::HostFactImporter
+    def import_facts(facts)
+        # Discovered Hosts won't report in via puppet, so we can use that field to
+        # record the last time it sent facts...
+        self.last_report = Time.now
+        # Set the correct facts type for new foreman facts importing code.
+        facts[:_type] = :foreman_discovery
+        super(facts)
+    end
+end
```
Comment (ezr-ondrej) [inline app/services/foreman_discovery/host_fact_importer.rb]: missing an trailing line
```suggestion
end

```

File: app/models/host/discovered.rb [hunk 2]
```diff
@@ -92,19 +96,11 @@ def self.import_host facts
 
     # and save (interfaces are created via puppet parser extension)
     host.save(:validate => false) if host.new_record?
-    raise ::Foreman::Exception.new(N_("Facts could not be imported")) unless host.import_facts(facts)
+    importer = ForemanDiscovery::HostFactImporter.new(host)
```
Comment (ezr-ondrej) [inline app/models/host/discovered.rb:99]:
```suggestion
    importer = ::ForemanDiscovery::HostFactImporter.new(host)
```

File: app/models/host/discovered.rb [hunk 3]
```diff
@@ -144,7 +140,7 @@ def proxy_url(node_ip)
   def refresh_facts
     facts = ::ForemanDiscovery::NodeAPI::Inventory.new(:url => proxy_url(self.ip)).facter
     self.class.import_host facts
-    import_facts facts
+    ForemanDiscovery::HostFactImporter.new(self).import_facts facts
```
Comment (ezr-ondrej) [inline app/models/host/discovered.rb]:
```suggestion
    ::ForemanDiscovery::HostFactImporter.new(self).import_facts facts
```

**Review by ezr-ondrej — APPROVED** [review]
Just nitpicks, general :+1:

File: app/services/foreman_discovery/host_fact_importer.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class ForemanDiscovery::HostFactImporter < ::HostFactImporter
+    def import_facts(facts)
+        # Discovered Hosts won't report in via puppet, so we can use that field to
+        # record the last time it sent facts...
+        self.last_report = Time.now
```
Comment (ezr-ondrej) [inline app/services/foreman_discovery/host_fact_importer.rb]:
```suggestion
        host.last_report = Time.now
```

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
Oh, this one I've missed.

**Comment by lzap** [comment]
Commited them all, I could not get my tests running locally yet so this was/is a blind try :-)

**Review by ezr-ondrej — APPROVED** [review]
Great :+1: WFM

**Pull request merged**